sdk/pinocchio/src/cpi.rs

```diff
@@ -50,34 +87,45 @@ pub fn invoke_signed<const ACCOUNTS: usize>(
     account_infos: &[&AccountInfo; ACCOUNTS],
     signers_seeds: &[Signer],
 ) -> ProgramResult {
-    if instruction.accounts.len() < ACCOUNTS {
-        return Err(ProgramError::NotEnoughAccountKeys);
-    }
```

Comment:
Isn't there still a chance that `instruction.accounts.len()` is greater than `ACCOUNTS`? Or are you relying on the syscall failing because we don't provide enough account infos for the instruction?

Reply:
Adding a test `instruction.accounts.len() != ACCOUNTS` at the start of the function increases CUs by `100`, so I am tempted to let the syscall fail instead.